From 4736a56c8e6ffee2f9101ab8d2b814eb239f5889 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Tue, 26 Sep 2017 07:26:02 +0000
Subject: [PATCH 1/1] abuild: do not try to strip standalone elf images

Guile uses ELF is internal object format, and creates them as
"no machine" and "standlone" OSABI. Scanelf supports printing
OSABI, so use that to filter these out.
See: https://github.com/alpinelinux/aports/pull/1714

This also removes unneccessary 'sed' from the pipeline as it's
simple to read each field outputted by scanelf.
---
 abuild.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/abuild.in b/abuild.in
index dae99b3..ef80891 100644
--- a/abuild.in
+++ b/abuild.in
@@ -2142,9 +2142,9 @@ stripbin() {
 	esac
 
 	msg "Stripping binaries"
-	scanelf --recursive --nobanner --etype "ET_DYN,ET_EXEC" . \
-		| sed -e 's:^ET_DYN ::' -e 's:^ET_EXEC ::' \
-		| while read filename; do
+	scanelf --recursive --nobanner --osabi --etype "ET_DYN,ET_EXEC" . \
+		| while read type osabi filename; do
+		[ "$osabi" != "STANDALONE" ] || continue
 		local XATTR=$(getfattr --match="" --dump "${filename}")
 		"${stripcmd}" "${filename}"
 		if [ -n "$XATTR" ]; then
-- 
2.14.1

